camel/personas/persona_generator.py

```diff
@@ -0,0 +1,230 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import os
+import re
+import uuid
+from typing import Dict, List, Optional, Union
+
+from camel.agents import ChatAgent
+from camel.configs import ChatGPTConfig
+from camel.messages import BaseMessage
+from camel.models import BaseModelBackend, ModelFactory
+from camel.personas import Persona
+from camel.prompts import TextPrompt
+from camel.types import ModelPlatformType, ModelType, RoleType
+
+
+class PersonaGenerator(ChatAgent):
+    r"""A generator of personas. This class manages a collection of Persona
+    instances and provides methods for adding, removing, and manipulating
+    personas within the group.
+
+    Args:
+        model (BaseModelBackend, optional): The model to use for persona
+        group_name (str, optional): The name of the group.
+        group_description (str, optional): A description of the group.
+    """
+
+    def __init__(
+        self,
+        model: Optional[BaseModelBackend] = None,
+    ):
+        system_message = BaseMessage(
+            role_name="",
+            role_type=RoleType.ASSISTANT,
+            meta_dict=None,
+            content="",
+        )
+        self.model = (
+            model
+            if model
+            else ModelFactory.create(
+                model_platform=ModelPlatformType.OPENAI,
+                model_type=ModelType.GPT_4O_MINI,
+                model_config_dict=ChatGPTConfig().__dict__,
+                api_key=os.getenv("OPENAI_API_KEY"),
+            )
+        )
+        super().__init__(system_message, model=model)
+        self.personas: Dict[uuid.UUID, Persona] = {}
+
+    def add_persona(self, persona: Persona):
+        r"""Add a persona to the group."""
+        self.personas[persona.id] = persona
+
+    def __delitem__(self, persona_id: uuid.UUID):
```

Comment:
since you are using __delitem__ and __getitem__ , use __setitem__ instead of add_persona to make the code style unified?